integration/suites/localregistry_test.go

```diff
@@ -0,0 +1,304 @@
+								Name:      "certs",
+								MountPath: "/certs/",
+							},
+						},
+					},
+				},
+				Volumes: []corev1.Volume{
+					{
+						Name: "certs",
+						VolumeSource: corev1.VolumeSource{
+							ConfigMap: &corev1.ConfigMapVolumeSource{
+								LocalObjectReference: corev1.LocalObjectReference{
+									Name: s.configMapName,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		metav1.CreateOptions{},
+	)
+	require.NoError(s.T(), err, "%s: create registry pod failed", s.Name)
+
+	logrus.Infof("%s: Creating ClusterIP Service for registry %s", s.Name, s.registryName)
+	_, err = s.serviceClient.Create(ctx,
+		&corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: s.registryName,
+			},
+			Spec: corev1.ServiceSpec{
+				Ports: []corev1.ServicePort{
+					{
+						Name: "registry-tls",
+						Port: 443,
+					},
+				},
+				Selector: map[string]string{
+					"app": s.registryName,
+				},
+				Type: corev1.ServiceTypeClusterIP,
+			},
+		},
+		metav1.CreateOptions{},
+	)
+	require.NoError(s.T(), err, "%s: create registry service failed", s.Name)
+
+	// Now create the builder with the certs mapped and a config that trusts this local registry
+	logrus.Infof("%s: Creating custom builder with registry cert and config %s", s.Name, s.registryName)
+	dir, cleanup, err := common.NewBuildContext(map[string]string{
+		"buildkitd.toml": fmt.Sprintf(`debug = false
+[worker.containerd]
+	namespace = "k8s.io"
+[registry."%s"]
+	ca=["/etc/config/cert.pem"]
+`, s.registryFQDN)})
+	require.NoError(s.T(), err, "%s: config file create failed", s.Name)
+
+	defer cleanup()
+
+	args := append(
+		[]string{
+			"--config", filepath.Join(dir, "buildkitd.toml"),
+			"--custom-config", s.configMapName,
+			s.Name,
+		},
+		s.CreateFlags...,
+	)
+	err = common.RunBuildkit("create", args)
+	require.NoError(s.T(), err, "%s: builder creation failed", s.Name)
+}
+
+func (s *localRegistrySuite) TearDownSuite() {
+	if !s.skipTeardown {
+
+		ctx := context.Background()
+
+		// Clean everything up...
+		err := s.podClient.Delete(ctx, s.registryName, metav1.DeleteOptions{})
+		if err != nil {
+			logrus.Warnf("failed to clean up pod %s: %s", s.registryName, err)
+		}
+		err = s.serviceClient.Delete(ctx, s.registryName, metav1.DeleteOptions{})
+		if err != nil {
+			logrus.Warnf("failed to clean up service %s: %s", s.registryName, err)
+		}
+		err = s.configMapClient.Delete(ctx, s.configMapName, metav1.DeleteOptions{})
+		if err != nil {
+			logrus.Warnf("failed to clean up configMap %s: %s", s.configMapName, err)
+		}
+		err = s.secretClient.Delete(ctx, s.Name, metav1.DeleteOptions{})
+		if err != nil {
+			logrus.Warnf("failed to clean up registry secret %s: %s", s.configMapName, err)
+		}
+
+		logrus.Infof("%s: Removing builder", s.Name)
+		err = common.RunBuildkit("rm", []string{
+			s.Name,
+		})
+		if err != nil {
+			logrus.Warnf("failed to clean up builder %s", err)
+		}
+
+	}
+}
+
+func (s *localRegistrySuite) TestBuildWithPush() {
```

Comment:
Not strictly necessary, but maybe a Pull as well too?

Reply:
Now that we have this baseline test rigging we should be able to add various test-cases on top  (caching, pull/push, etc.)  - lets do those as additive changes.